npm/qsharp/ux/circuit-vis/contextMenu.ts

```diff
@@ -325,33 +358,24 @@ const _createInputPrompt = (
     validateAndToggleOkButton();
   });
 
-  // Create the OK button
-  const okButton = document.createElement("button");
-  okButton.classList.add("prompt-button");
-  okButton.textContent = "OK";
-  okButton.disabled = !validateInput(defaultValue);
-  okButton.addEventListener("click", () => {
-    callback(inputElem.value.trim());
-    document.body.removeChild(overlay);
-  });
-
   // Create the Cancel button
   const cancelButton = document.createElement("button");
   cancelButton.classList.add("prompt-button");
   cancelButton.textContent = "Cancel";
   cancelButton.addEventListener("click", () => {
     callback(null);
     document.body.removeChild(overlay);
+    document.removeEventListener("keydown", handleGlobalKeyDown, true);
   });
 
-  // Function to validate input and toggle the OK button
-  const validateAndToggleOkButton = () => {
-    const isValid = validateInput(inputElem.value.trim());
-    okButton.disabled = !isValid;
+  // Handle Escape key globally while prompt is open
+  const handleGlobalKeyDown = (event: KeyboardEvent) => {
+    if (event.key === "Escape") {
+      event.preventDefault();
+      cancelButton.click();
+    }
   };
```

Comment:
Does this need to be global here? There is already a global handler defined in events.ts, and the previous one in this file is for key events on the input field. Not sure if this is needed or redundant.

Reply:
This is needed because the other one defined in this file, for the input prompt, as well as this one are both temporary handlers. They are specific to the prompt that creates them and will be removed globally when the prompt is removed. Even if we didn't remove them, we'd still need to create new ones each time because their logic is specific to the prompt instance that creates them and removing them just makes sure to not "leak" orphaned event handlers on the document.